utils/anautilities.py

```diff
@@ -55,7 +55,7 @@ def dacAnalysis(args, dacScanTree, chamber_config, scandate='noscandate'):
     dict_nonzeroVFATs = {}
     for entry in crateMap:
         ohKey = (entry['shelf'],entry['slot'],entry['link'])
-        dict_nonzeroVFATs[ohKey] = np.unique(vfatArray[np.logical_and(vfatArray['dacValY'] > 0,vfatArray['link'] == ohKey)]['vfatN'])
+        dict_nonzeroVFATs[ohKey] = np.unique(vfatArray[np.logical_and(vfatArray['dacValY'] > 0,vfatArray['link'] == entry['link'])]['vfatN'])
```

Comment:
this should then probably include a check on shelf and slot as well to ensure the correct array is paired with the correct `ohKey`.

Reply:
Probably the following would work:

```python
    # get nonzero VFATs                                                                                                                                                                                                            
    dict_nonzeroVFATs = {}
    for entry in crateMap:
        arrayMask = np.logical_and(
                vfatArray['dacValY'] > 0,
                vfatArray['link'] == entry['link'],
                vfatArray['slot'] == entry['slot'],
                vfatArray['shelf'] == entry['shelf'],
                )
        dict_nonzeroVFATs[ohKey] = np.unique(vfatArray[arrayMask]['vfatN'])
```

Reply:
The actual solution looks like:

```python
    # get nonzero VFATs
    dict_nonzeroVFATs = {}
    for entry in crateMap:
        ohKey = (entry['shelf'],entry['slot'],entry['link'])
        arrayMask = np.logical_and(vfatArray['rate'] > 0, vfatArray['link'] == entry['link'])                                                                                                                                    
        arrayMask = np.logical_and(arrayMask, vfatArray['slot'] == entry['slot'])
        arrayMask = np.logical_and(arrayMask, vfatArray['shelf'] == entry['shelf'])
        dict_nonzeroVFATs[ohKey] = np.unique(vfatArray[arrayMask]['vfatN'])
```

`np.logical_and(...)` doesn't seem smart enough to take multiple arrays.  One could do it in two lines by taking the logical and of two logical ands but that I feel would degrade readability.

Reply:
@AndrewLevin ?

Reply:
I just pushed your solution, but haven't tested it (`testConnectivity.py` does not seem to save the DAC scan data, so I need to run another one).